src/tests/ftest/launch.py

```diff
@@ -925,50 +939,75 @@ def archive_config_files(avocado_logs_dir):
     archive_files(
         destination, host_list, "{}/*_*_*.yaml".format(configs_dir))
 
-
-def archive_files(destination, host_list, source_files):
+def archive_files(destination, host_list, source_files, do_tar=False):
     """Archive all of the remote files to the destination directory.
 
     Args:
         destination (str): path to which to archive files
         host_list (list): hosts from which to archive files
         source_files (str): remote files to archive
+        do_tar (bool): tar directory before scp-ing to host
+          (Useful for copying, e.g., files from orterun --output-filename option
+          to the avocado results directory.)
+
     """
     this_host = socket.gethostname().split(".")[0]
 
     # Create the destination directory
-    get_output(["mkdir", destination])
+    if not os.path.exists(destination):
+      get_output(["mkdir", destination])
 
     # Display available disk space prior to copy.  Allow commands to fail w/o
     # exiting this program.  Any disk space issues preventing the creation of a
     # directory will be caught in the archiving of the source files.
     print("Current disk space usage of {}".format(destination))
     print(get_output(["df", "-h", destination]))
 
+    # If we're tar-ing, then we're our source files ought to be a directory,
+    # hence the -d option to ls here.
+    ls_cmd = "ls"
+    if do_tar:
+      ls_cmd = "ls -d"
+
     # Copy any source files that exist on the remote hosts and remove them from
     # the remote host if the copy is successful.  Attempt all of the commands
     # and report status at the end of the loop.  Include a listing of the file
     # related to any failed command.
     commands = [
         "set -eu",
+        "set +x",
```

Comment:
Not sure this is required as we are not turning on traces (-e).  In general though it is useful for debugging to have traces enabled to see if there are any issues with variable substitution.